flink-java/src/main/java/org/apache/flink/api/java/functions/SelectByMinFunction.java

```diff
@@ -72,8 +73,8 @@ public T reduce(T value1, T value2) throws Exception {
 		for (int position : fields) {
 			// Save position of compared key
 			// Get both values - both implement comparable
-			Comparable comparable1 = value1.getFieldNotNull(position);
-			Comparable comparable2 = value2.getFieldNotNull(position);
+			Comparable comparable1 = ((Tuple)value1).getFieldNotNull(position);
```

Comment:
I think the purpose of Flink-3650 is adding support for Scala tuples. Thus, I would rather drop support for Java tuples in the Scala API than for Scala tuples. I would assume that you have to implement a Scala specific `SelectByMax/MinFunction` to support Scala tuples.